### PR TITLE
[WIP] Custom contract Id for published contract

### PR DESCRIPTION
**Pull request by kumaryash90**
## PR-Codex overview
This PR updates the SDK to allow custom contract IDs for published contracts.

### Detailed summary
- Added `contractId` field with regex validation in `legacy_packages/sdk/src/evm/schema/contracts/custom.ts`
- Updated `contractId` logic in `legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts`
- Enhanced UI with `Contract ID` input field in form components.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> /release-pr

this is no longer a thing

**Comment by kumaryash90** [comment]
> > /release-pr
> 
> this is no longer a thing

how do we create dev builds now? @jnsdls

File: legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts [hunk 1]
```diff
@@ -486,7 +486,7 @@ export class ContractPublisher extends RPCConnectionHandler {
         : `0x${fetchedBytecode}`;
 
       const bytecodeHash = utils.solidityKeccak256(["bytes"], [bytecode]);
-      const contractId = predeployMetadata.name;
+      const contractId = extraMetadata.contractId || predeployMetadata.name;
```
Comment (joaquim-verges) [inline legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts:489]: make sure there isnt any other places we use contractId - i know we use it as the key for a lot of things.

Also can you do the change in v5 as well?

Comment (kumaryash90) [reply legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts:489]: where can I find the v4 ContractPublisher class equivalent in v5?

**Comment by kumaryash90** [comment]
Closing this PR. 

Making the Contract ID override change directly on CLI instead: https://github.com/thirdweb-dev/js/pull/3424.

**Comment by kumaryash90** [comment]
Closing this PR.

Making the Contract ID override change directly on CLI instead: https://github.com/thirdweb-dev/js/pull/3424.

**Pull request closed without merging**